src/Button/Button.component.tsx

```diff
@@ -3,113 +3,602 @@ import * as React from 'react';
 // VENDOR
 import classNames from 'classnames';
 import styled, { css, FlattenSimpleInterpolation } from 'styled-components';
-import { darken } from 'polished';
+import { transparentize } from 'polished';
 // THEME
 import { colors, fonts, sizes } from '../theme';
+import { Colors } from '../Typography/Typography.component';
+
+type Variant = 'primary' | 'outline' | 'minimal';
+
+type ButtonSize = 'xlarge' | 'large' | 'small' | 'xsmall';
+
+interface Theme {
+    base: string;
+    light: string;
+    dark: string;
+}
 
 interface ButtonProps {
     children?: any;
     disabled?: boolean;
+    revealed?: boolean;
     showSpinner?: boolean;
     className?: string;
 
+    flip?: boolean;
+    colorTheme?: Theme | Colors;
+    minWidth?: string;
+    block?: boolean;
+    icon?: any;
+    circular?: boolean;
+    reverse?: boolean;
+
+    forceHover?: boolean;
+    forceFocus?: boolean;
+    forceActive?: boolean;
```

Comment:
I get these for a storybook story, but I wonder if we'll ever need these for any use-case other than that? If not, maybe there's another way we can implement this.

Reply:
I agree (my above comment:)

>Is there any way to make this functionality's code (forceActive, forceFocus, forceHover) dev only?

Reply:
Hah, missed that. What about using something as primitive as classes?

Reply:
I thought about this solution for awhile- the issue I'm have with using classes is that the dom state styles to apply with `classes` are a function of a bunch of the button's props (color, circular etc).

I refactored it a little at least so that all of the `force*` css is behind a single `(forceHover || forceFocus || forceActive)` condition.

Reply:
I don't see a hard reason to keep the props, but I don't think it's completely out of the realm of possibility that someone might want to force any of those states. If anything it provides more programmatic options and that's not a bad thing. As long as it's documented well, I say keep them. That said, add a comment to this effect in the code.